dynamodb/src/main/java/org/apache/metamodel/dynamodb/DynamoDbTableCreationBuilder.java

```diff
@@ -0,0 +1,111 @@
+import com.amazonaws.services.dynamodbv2.model.KeyType;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughput;
+import com.amazonaws.services.dynamodbv2.model.TableStatus;
+
+class DynamoDbTableCreationBuilder extends AbstractTableCreationBuilder<DynamoDbUpdateCallback> {
+
+    private static final Logger logger = LoggerFactory.getLogger(DynamoDbTableCreationBuilder.class);
+
+    public DynamoDbTableCreationBuilder(DynamoDbUpdateCallback updateCallback, Schema schema, String name) {
+        super(updateCallback, schema, name);
+    }
+
+    @Override
+    public Table execute() throws MetaModelException {
+        final MutableTable table = getTable();
+        final String tableName = table.getName();
+
+        final Collection<AttributeDefinition> attributes = new ArrayList<>();
+        final Collection<KeySchemaElement> keySchema = new ArrayList<>();
+        final Collection<GlobalSecondaryIndex> globalSecondaryIndices = new ArrayList<>();
+
+        final long readCapacity = Long.parseLong(System.getProperty(
+                DynamoDbDataContext.SYSTEM_PROPERTY_THROUGHPUT_READ_CAPACITY, "5"));
+        final long writeCapacity = Long.parseLong(System.getProperty(
+                DynamoDbDataContext.SYSTEM_PROPERTY_THROUGHPUT_WRITE_CAPACITY, "5"));
+        final ProvisionedThroughput provisionedThroughput = new ProvisionedThroughput(readCapacity, writeCapacity);
+
+        final Column[] columns = table.getColumns();
+        for (Column column : columns) {
+            if (column.isPrimaryKey()) {
+                final KeyType keyType = getKeyType(column.getRemarks());
+                keySchema.add(new KeySchemaElement(column.getName(), keyType));
+                attributes.add(new AttributeDefinition(column.getName(), DynamoDbUtils.toAttributeType(column
+                        .getType())));
+            }
+        }
+
+        final CreateTableRequest createTableRequest = new CreateTableRequest();
+        createTableRequest.setTableName(tableName);
+        createTableRequest.setAttributeDefinitions(attributes);
+        createTableRequest.setGlobalSecondaryIndexes(globalSecondaryIndices);
+        createTableRequest.setKeySchema(keySchema);
+        createTableRequest.setProvisionedThroughput(provisionedThroughput);
+
+        final AmazonDynamoDB client = getUpdateCallback().getDataContext().getDynamoDb();
+
+        final CreateTableResult createTableResult = client.createTable(createTableRequest);
+
+        // await the table creation to be "done".
+        {
+            String tableStatus = createTableResult.getTableDescription().getTableStatus();
+            while (TableStatus.CREATING.name().equals(tableStatus)) {
+                logger.debug("Waiting for table status to be ACTIVE. Currently: {}", tableStatus);
+                try {
+                    Thread.sleep(300);
+                } catch (InterruptedException e) {
+                }
```

Comment:
But it would be someone "interrupting our sleep", not interrupting the thread in general, would it? Basically the table creation on Amazon takes some time - I've seen it taking from 1 second to 10 seconds. But since you can't do anything with the table in that timespan, I wanted a sleeping checker to wait before we return. An interruption of that ... Hmm I can't even see what that would mean :)